src/App.tsx

```diff
@@ -1,24 +1,228 @@
+import { Notification } from './components/Notification';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { UserWarning } from './UserWarning';
+import { getActiveTodos } from './utils/getActiveTodos';
+import { getCompletedTodos } from './utils/getCompletedTodos';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [hasError, setHasError] = useState(false);
+  const [error, setError] = useState<ErrorType>(ErrorType.NO_ERROR);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [activeTodosId, setActiveTodosId] = useState<number[]>([]);
+
+  const USER_ID = 6381;
+
+  const getAllTodos = async (userId: number) => {
+    try {
+      const todosFromServer = await getTodos(userId);
+
+      setTodos(todosFromServer);
+    } catch {
+      setHasError(true);
+      setError(ErrorType.UPLOAD_ERROR);
+      // eslint-disable-next-line no-console
+      console.warn('An occur error while loading todos');
+    }
+  };
+
+  useEffect(() => {
+    getAllTodos(USER_ID);
+  }, []);
+
+  const activeTodos = useMemo(() => getActiveTodos(todos), [todos]);
+
+  const completedTodos = useMemo(() => getCompletedTodos(todos), [todos]);
+
+  const handleAddTodo = async (userId: number, title: string) => {
+    if (!title.trim()) {
+      setHasError(true);
+      setError(ErrorType.EMPTY_ERROR);
+
+      return;
+    }
+
+    const newTodo = {
+      id: 0,
+      userId,
+      title,
+      completed: false,
+    };
+
+    try {
+      setIsInputDisabled(true);
+      setTempTodo(newTodo);
+      await addTodo(newTodo);
+      await getAllTodos(USER_ID);
+    } catch {
+      setHasError(true);
+      setError(ErrorType.ADD_ERROR);
+      // eslint-disable-next-line no-console
+      console.warn('An occur error while adding todo');
+    } finally {
+      setIsInputDisabled(false);
+      setTempTodo(null);
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    setActiveTodosId(prevTodosId => [
+      ...prevTodosId,
+      todoId,
+    ]);
+
+    try {
+      await deleteTodo(todoId);
+      await getAllTodos(USER_ID);
+    } catch {
+      setHasError(true);
+      setError(ErrorType.DELETE_ERROR);
+      // eslint-disable-next-line no-console
+      console.warn('An occur error while deleting todo');
+    } finally {
+      setActiveTodosId(prevTodosId => prevTodosId.filter(id => id !== todoId));
+    }
+  };
+
+  const deleteCompletedTodos = useCallback(() => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        handleDeleteTodo(todo.id);
+      }
+    });
+  }, [todos]);
+
+  const changeTodo = async (todo: Todo, title?: string) => {
+    if (todo.title === title) {
+      return;
+    }
+
+    if (title?.length === 0) {
```

Comment:
```suggestion
    if (!title?.length) {
```

Reply:
I left `title?.length === 0` because title can be `undefined`, and `!title.length` would work incorrectly, so I needed to add an additional check for `undefined`.
I think first variant is more clear)